tensorboard/webapp/metrics/views/right_pane/settings_view_container.ts

```diff
@@ -44,6 +48,14 @@ import {
   XAxisType,
 } from '../../types';
 
+const RANGE_INPUT_SOURCE_TO_AFFORDANCE: Record<
```

Comment:
This object seems needlessly complicated. Why not just use source === RangeInputSource.TEXT ? TimeSelectionAffordance.SETTINGS_TEXT | TimeSelectionAffordance.SETTINGS_SLIDER.  That seems easier to read to me but maybe that is just my opinion.

That being said I do not see anything wrong with this way of converting.

Reply:
This will lead to a loud failure if we add to `RangeInputSource` which I view as desirable.